pwiz_tools/Shared/ProteowizardWrapper/MsDataFileImpl.cs

```diff
@@ -1793,8 +1794,17 @@ private static IEnumerable<MsPrecursor> GetMs1Precursors(Spectrum spectrum)
 
         private static CVID? GetPrecursorDissociationMethod(Precursor precursor)
         {
-            return precursor.activation?.cvParams
-                .FirstOrDefault(param => CV.cvIsA(param.cvid, CVID.MS_dissociation_method))?.cvid;
+            foreach (var cvParam in (IEnumerable<CVParam>) precursor.activation.cvParams ?? Array.Empty<CVParam>())
+            {
+                using (cvParam)

```

Comment:
Hah I didn't even know you could do that. Not sure why this is still leaking. But I did realize this could be more efficient. Use: `precursor.activation.cvParamChildren(CVID.MS_dissociation_method)` to get the list of CVParams falling under that term, you don't have to do the matching yourself. Note I switched to the plural version because there can be more than one method for a precursor. For example, EThcD will have HCD and ETD (HCD is a "supplemental" activation). So perhaps the method will need to return List<CVID>.

Reply:
@chambm, do you have an example of a file with this sort of multiple dissociation methods? I'll add it to my unit test.

The dissociation method ends up being a string by the time Skyline sees it.
If there is more than one dissociation method in a single spectrum (which could theoretically happen if there was more than one precursor) then the value ends up being the methods separated by spaces. That might be how we would want this to work. Since the value is just a string, users could use "contains" in the Spectrum Filter that they define.

Reply:
We may want to special case this such that if supplemental CID is there, then regular CID is omitted. Or perhaps vice versa if we think users would rather stick with the short names.